methods.go

```diff
@@ -2,10 +2,34 @@ package telego
 
 import (
 	"fmt"
+	"math"
+	"time"
 
 	"github.com/mymmrac/telego/telegoapi"
 )
 
+// `retry` - executes a function multiple times and returns the final `err` value.
+// It finishes when either `err == nil` or `maxAttempts` is reached.
+func (b *Bot) retry(fn func() error) error {
+	var err error
+
+	for i := 0; i < b.retryOptions.maxAttempts; i++ {
+		err = fn()
+		if err == nil {
```

Comment:
Here it would be nice to check if the error is from Telegram itself, for things like bad request there is no need to retry request (the exact error codes should be checked, I don't know if Telegram uses HTTP conventions for them, or it's something else)

To check if error is from Telegram use `ta.Error` type (`telegoapi`)